docker-compose.yml

```diff
@@ -0,0 +1,92 @@
+version: "2.1"
```

Comment:
why such an old version of the compose file?

Reply:
The compose file is making use of a healthcheck feature that they removed in version 3. It's built in a way that none of the microservices start until the config service returns a 200.

Reply:
what's missing from here? https://docs.docker.com/compose/compose-file/compose-file-v3/#healthcheck

Reply:
The services named accounts, commerce and inventory each depend on config's condition equaling 'service_healthy' (e.g line 56). Docker compose version 3 removed this. [the second example at this link shows this feature](https://docs.docker.com/compose/compose-file/compose-file-v2/#depends_on). 

What happens without that healthcheck is all three of those services run and fail about 5 times before they're ready because they can't get info from config, which makes the logs very difficult to read.

It's similar to what they did in their startup.sh (line 17-20). 

Further, the latest docker-compose and docker version are backwards compatible with version 2.1.

Reply:
yes, docker compose removed this feature because it just wasn't good enough at it for all edge cases and didn't want to try to be. on the other hand, it looks like since ~10 months ago, they effectively added it back in by moving away from the "version 3" format to a "unified" format.
https://docs.docker.com/compose/compose-file/compose-versioning/
based on this, i'm no longer worried about an old version of the format since apparently backwards compatibility is being guaranteed. you might consider removing the `version:` as it is deprecated apparently.